aiven/client/connection_info/kafka.py

```diff
@@ -88,19 +104,25 @@ def from_service(
             raise ConnectionInfoError(
                 "Cannot format kafka connection info for service type {service_type}".format_map(service)
             )
+        
+        try:
+            info = find_component(
+                service["components"],
+                route=route,
+                privatelink_connection_id=privatelink_connection_id,
+                kafka_authentication_method="sasl",
+            )
+            
+            user = find_user(service, username)
+            if "password" not in user:
+                raise ConnectionInfoError(f"Could not find password for username {username}")
+            return cls(host=info["host"], port=info["port"], username=username, password=user["password"])
+        except ConnectionInfoError:
```

Comment:
If the password is missing, then you'll get `SASL auth...` error, which is wrong. I think the better way is to check user config via separate call to understand if SASL auth is enabled.

Reply:
gotcha, doing for both SSL and SASL methods